pkgs/development/python-modules/qubes-core-qrexec/default.nix

```diff
@@ -0,0 +1,140 @@
+    sourceRoot = "${src.name}/agent";
+
+    nativeBuildInputs = [
+      pkg-config
+      pandoc
+    ];
+
+    buildInputs = [
+      qubes-core-vchan-xen
+      util-agent-bootstrap
+    ];
+
+    postInstall = ''
+      mv $out/usr/* $out/
+      rm -rf $out/usr
+      # Scripts: qubes-rpc-multiplexer
+      cp ../lib/* $out/bin/
+    '';
+
+    makeFlags = [ "DESTDIR=${placeholder "out"}" ];
+  };
+  # Util hardcodes agent paths, but it is used by agent.
+  # Building separate package for agent, which doesn't use those paths.
+  util-hack = agent: stdenv.mkDerivation {
+    inherit version src;
+    pname = "qubes-core-qrexec-util";
+    sourceRoot = "${src.name}/libqrexec";
+
+    # TODO: qrexec-policy-exec is located in qubes-core-qrexec, thus can't be unhardcoded this easy...
+    # Maybe all of those packages should be built as one, but it is complicated.
+    # As a temporary solution, I think persistent name should be good enough?..
+    patchPhase = ''
+      substituteInPlace qrexec.h \
+        --replace-fail "/var/run/qubes/qrexec-agent-fdpass" "${agent}/bin/qrexec-agent-fdpass" \
+        --replace-fail "/var/run/qubes/qrexec-agent" "${agent}/bin/qrexec-agent" \
+        --replace-fail "/usr/lib/qubes/qubes-rpc-multiplexer" "${agent}/bin/qubes-rpc-multiplexer" \
+        --replace-fail "/usr/bin/qrexec-policy-exec" "/run/current-system/sw/bin/qrexec-policy-exec"
+    '';
+
+    nativeBuildInputs = [
+      pkg-config
+    ];
+
+    buildInputs = [
+      qubes-core-vchan-xen
+    ];
+
+    makeFlags = [ "DESTDIR=${placeholder "out"}" "LIBDIR=/lib" "INCLUDEDIR=/include" ];
+  };
+  util-agent-bootstrap = (util-hack "I AM UNUSED").overrideAttrs (oldAttrs: {
+    pname = "${oldAttrs.pname}-agent-bootstrap";
+  });
+  util = util-hack agent;
+  daemon = stdenv.mkDerivation {
+    inherit version src;
+    pname = "qubes-core-qrexec-daemon";
+    sourceRoot = "${src.name}/daemon";
+
+    nativeBuildInputs = [
+      pkg-config
+    ];
+
+    buildInputs = [
+      qubes-core-vchan-xen
+      util
+    ];
+
+    postInstall = ''
+      mv $out/usr/* $out/
+      mv $out/sbin/* $out/bin/
+      rm -rf $out/usr $out/sbin
+    '';
+
+    makeFlags = [ "DESTDIR=${placeholder "out"}" ];
+  };
+in
+
+buildPythonPackage {
+  inherit version src;
+  pname = "qubes-core-qrexec";
+  format = "setuptools";
+
+  patchPhase = ''
+    substituteInPlace qrexec/client.py \
+      --replace-fail "/usr/bin/qrexec-client-vm" "${agent}/bin/qrexec-client-vm" \
+      --replace-fail "/usr/lib/qubes/qubes-rpc-multiplexer" "${agent}/bin/qubes-rpc-multiplexer" \
+      --replace-fail "/usr/bin/qrexec-client" "${daemon}/lib/qubes/qrexec-client" \
+  '';
+
+  passthru = {
+    # FIXME: Should be split into separate packages?
+    inherit agent util daemon;
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook3
+  ];
+
+  propagatedBuildInputs = [
+    pyinotify
+    pygobject3
+    gtk3
+  ];
+
+  # Should also be packaged as vm-only?
```

Comment:
A split output package sounds good here IMO.

Reply:
I think current approach with passthru is okayish, as those packages are tightly bound.
Proper splitting will require a qubes package set, which I want to avoid.